test/e2e/integration_runtime_test.go

```diff
@@ -135,22 +135,53 @@ func TestHostSocketConnect(t *testing.T) {
 	}
 	defer unix.Close(tmpDirFD)
 	// Use /proc/self/fd to generate path to avoid EINVAL on large path.
-	l, err := net.Listen("unix", filepath.Join("/proc/self/fd", strconv.Itoa(tmpDirFD), "test.sock"))
+	socketPath := filepath.Join("/proc/self/fd", strconv.Itoa(tmpDirFD), "test.sock")
+	l, err := net.Listen("unix", socketPath)
 	if err != nil {
 		t.Fatalf("listen error: %v", err)
 	}
 	defer l.Close()
+	// change the socket's permission so that "nonroot" can connect to it
+	if err := os.Chmod(socketPath, 0777); err != nil {
+		t.Errorf("chmod error: %v", err)
+	}
 
 	var wg sync.WaitGroup
 	wg.Add(1)
 	go func() {
+		checkPeerCreds := func(conn net.Conn) bool {
+			unixConn, ok := conn.(*net.UnixConn)
+			if !ok {
+				t.Fatalf("expected *net.UnixConn, got %T", conn)
+				return false
+			}
+			file, err := unixConn.File()
+			if err != nil {
+				t.Errorf("file error: %v", err)
+				return false
+			}
+			defer file.Close()
+			cred, err := unix.GetsockoptUcred(int(file.Fd()), unix.SOL_SOCKET, unix.SO_PEERCRED)
+			if err != nil {
+				t.Errorf("getsockopt error: %v", err)
+				return false
+			}
+			if cred.Uid != 1337 || cred.Gid != 1337 {
```

Comment:
Might be useful to define 1337 as nonRootUID and nonRootGID constants. In case future tests using "nonroot" also need this.